src/splinefunctions.cpp

```diff
@@ -0,0 +1,883 @@
+                  "spline node, but spline has been specified not to allow "
+                  "extrapolation.");
+
+            case SplineExtrapolation::constant:
+                return coefficients_extrapolate[2];
+
+            case SplineExtrapolation::linear:
+                return coefficients_extrapolate[2] +
+                       t * coefficients_extrapolate[3];
+
+            case SplineExtrapolation::polynomial:
+                /* Evaluate last interpolation polynomial */
+                i_node = n_nodes() - 2;
+                len = nodes_[i_node + 1] - nodes_[i_node];
+                return evaluatePolynomial((t - nodes_[i_node]) / len,
+                                          &(coefficients[i_node * 4]));
+
+            case SplineExtrapolation::periodic:
+                len = nodes_[n_nodes() - 1] - nodes_[0];
+                return get_value(nodes_[0] + std::fmod(t - nodes_[0], len));
+            default:
+                throw AmiException("Unsupported SplineExtrapolation type");
+        }
+    }
+
+    /* Are we before the first node? Extrapolate! */
+    if (t < nodes_[0]) {
+        switch (first_node_ep_) {
+            case SplineExtrapolation::noExtrapolation:
+                throw AmiException(
+                  "Trying to evaluate spline before first "
+                  "spline node, but spline has been specified not to allow "
+                  "extrapolation.");
+
+            case SplineExtrapolation::constant:
+                return coefficients_extrapolate[0];
+
+            case SplineExtrapolation::linear:
+                return coefficients_extrapolate[0] +
+                       t * coefficients_extrapolate[1];
+
+            case SplineExtrapolation::polynomial:
+                /* Evaluate last interpolation polynomial */
+                len = nodes_[1] - nodes_[0];
+                return evaluatePolynomial((t - nodes_[0]) / len,
+                                          &(coefficients[0]));
+
+            case SplineExtrapolation::periodic:
+                len = nodes_[n_nodes() - 1] - nodes_[0];
+                return get_value(nodes_[n_nodes() - 1] +
+                                 std::fmod(t - nodes_[0], len));
+            default:
+                throw AmiException("Unsupported SplineExtrapolation type");
+        }
+    }
+
+    /* Get the spline interval which we need */
+    if (get_equidistant_spacing()) {
+        /* equidistant spacing: just compute the interval */
+        len = nodes_[1] - nodes_[0];
+        i_node = trunc((t - nodes_[0]) / len);
+    } else {
+        /* no equidistant spacing: we need to iterate */
+        i_node = 0;
+        while (nodes_[i_node + 1] < t) {
+            i_node++;
+        }
+        len = nodes_[i_node + 1] - nodes_[i_node];
+    }
+
+    /* Evaluate the interpolation polynomial */
+    return evaluatePolynomial((t - nodes_[i_node]) / len,
+                              &(coefficients[i_node * 4]));
```

Comment:
@lcontento It seems there is some out-of-bounds read in `coefficients` here. Please check

Reply:
I think there was an edge case (t = final point) which was not taken care of. But even correcting for that I still get segfaults.

Reply:
Seems resolved.